cpp/src/parquet/arrow/schema.cc

```diff
@@ -382,18 +382,18 @@ Status FieldToNode(const std::string& name, const std::shared_ptr<Field>& field,
       break;
     case ArrowTypeId::TIME32:
       type = ParquetType::INT32;
-      logical_type =
-          LogicalType::Time(/*is_adjusted_to_utc=*/true, LogicalType::TimeUnit::MILLIS);
+      logical_type = LogicalType::Time(arrow_properties.time_is_adjusted_to_utc(),
```

Comment:
Unlike the `TIMESTAMP` type, `TIME32` and `TIME64` types do not carry timezone info. Therefore reading a `Time32Array` back has no way to warn the reader that values are not adjusted to UTC in the parquet file.

Reply:
I see, this makes sense.  

Do you think it would make sense to add the `is_adjusted_to_utc` field to the TIME32/TIME64 data types, instead of having it passed as a writer property?  This would also allow specifying per-column, but I think it would be quite a large change.

Alternatively, it might be sufficient to log a warning during the schema conversion on read if a time32/time64 type is not adjusted to UTC?  Perhaps in `NodeToSchemaField` or `GetTypeForNode`?

Reply:
It might also be worth noting that a user can use `pq.ParquetFile.schema` in python or use the `ParquetFileReader`'s metadata in c++ to get the `is_adjusted_to_utc` field from the parquet schema (even though its not exposed in the arrow schema or TIME32 or TIME64 arrow types)

Reply:
IMO, we need to add `TIMETZ` in addition to `TIME` type, like what DuckDB has done: https://duckdb.org/docs/sql/data_types/time.html

WDYT? @pitrou @jorisvandenbossche

Reply:
As I said below, I think this flag makes no sense for times. It's ok to write it to Parquet for compatibility, but all times are conceptually local to an unspecified timezone, with no possibility to compute an equivalent "UTC time" since you don't know on which day is was measured.

Reply:
I have the same feeling that it makes no sense. Perhaps we should deprecate the flag on the parquet side? TBH, I'm not very clear on its use case, though the discussion in the issue and the DuckDB implementation seems provide some (valid) use cases.

Reply:
Perhaps this can be discussed on dev@parquet?

Reply:
Agreed. Let me do more research before that.

Reply:
@pitrou Responding to your other comment here.  The use case for `is_adjusted_to_utc=false` is mostly for compatibility reasons, as you mentioned above.  We have other parquet writers writing with `is_adjusted_to_utc=false`, and we'd like our arrow writers to produce parquet files with the same logical schema.

After thinking about it, I also agree that having a time type with a time zone (or time zone adjustment, as is the case with `is_adjusted_to_utc=true`) is not well defined.  I believe that one of the original DuckDB [discussions](https://github.com/duckdb/duckdb/discussions/2647) says that `The TIME WITH TIME ZONE is not well-defined because binning a time requires a specific date as well as a time zone. We could decide to add this type for completeness...`, so it seems like it was added solely for completeness. 

One possible use case I could see is if someone wanted to store a dataset of events that happen every day at the same time (for example, reporting deadlines that need to be met every day).  Then the expectation might be that readers would read the time and interpret it as an instant on the current day.  But if readers are in different time zones, they need to interpret the times as the same instant.

However, even in that case it might be better to store a `timestamptz` type and have a different event for each day, so the example is a bit contrived.

Regardless, would it be ok to add a `false` option to arrow for compatibility purposes and discuss deprecating it from parquet separately on dev@parquet?

Reply:
It seems that ANSI SQL 92 has also defined `TIME WITH TIME ZONE` type which is pretty controversial: https://stackoverflow.com/questions/60491608/does-the-sql-type-time-with-timezone-make-sense